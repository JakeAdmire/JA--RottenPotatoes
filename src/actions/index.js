export const setRestaurants = (restaurants) => ({
  type: 'SET_RESTAURANTS',
  restaurants
})

export const setLocation = (location) => ({
  type: 'SET_LOCATION',
  location
})

export const isLoading = (boolean) => ({
  type: 'IS_LOADING',
  boolean
})

export const setCard = (card) =>({
  type: 'SET_CARD',
  card
})