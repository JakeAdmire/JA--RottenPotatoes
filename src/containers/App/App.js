import React, { Component } from 'react';

class App extends Component {

  componentDidMount() {
    this.fetchRestaurants();
  }

  async fetchRestaurants() {
    const url = 'https://data.colorado.gov/resource/d5e8-gubm.json';
    try {
      const response = await fetch(url);
      const data = await response.json();
      this.gatherRestaurantNames(data);
    } catch(error) {
      console.log(error);
    }
  }

  gatherRestaurantNames = (data) => {
    const facilities = data.map(facility =>  facility.facilityname)
    const flatFacilities = [...new Set(facilities)];
    // this.props.setRestaurants(flatFacilities);
  }

  render() {
    return (
      <div className="App">
        <h1>Rotten Potatoes</h1>
      </div>
    );
  }
}

export default App;
