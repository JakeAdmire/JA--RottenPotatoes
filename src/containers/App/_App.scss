.App {
  font-family: 'Lobster', cursive;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.background-image {
  position: absolute;
  z-index: -2;
  width: 100%; height: 100%;
  background-image: url('./media/background.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  filter: blur(5px) opacity(.8);
  transform: scale(1.1)
}