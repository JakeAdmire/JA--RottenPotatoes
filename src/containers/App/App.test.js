import React from 'react';
import { shallow } from 'enzyme';

import { App, mapStateToProps, mapDispatchToProps } from './App';
import { setRestaurants } from '../../actions';

describe('App', () => {

  let wrapper;
  let mockSetRestaurants = jest.fn()
  let mockFetchRestaurants = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <App  setRestaurants={mockSetRestaurants}
            fetchRestaurants={mockFetchRestaurants} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke the fetchRestaurants function on componentDidMount', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'fetchRestaurants')
    wrapper.instance().componentDidMount();
    expect(instance.fetchRestaurants).toBeCalled();
  })

  it('should dispatch setRestaurants when gatherRestaurantNames is called', () => {
    let mockData = [
      { facilityname: 'East Moon Bistro' },
      { facilityname: 'Taco Johns' },
    ]
    wrapper.instance().gatherRestaurantNames(mockData);
    expect(mockSetRestaurants).toHaveBeenCalled();
  })

  describe('mapStateToProps', () => {

    it('should return a props object', () => {
      let mockState = { redirect: false };
      let results = mapStateToProps(mockState);
      expect(results).toEqual(mockState);
    })

  })

  describe('mapDispatchToProps', () => {

    it('should dispatch setRestaurants when its prop is called', () => {
      let mockRestaurants = [{}, {}];
      const mockDispatch = jest.fn();
      const actionToDispatch = setRestaurants(mockRestaurants);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setRestaurants(mockRestaurants);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })  

  })

})