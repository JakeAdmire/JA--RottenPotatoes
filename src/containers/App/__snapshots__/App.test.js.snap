// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchRestaurants={[MockFunction]}
    setRestaurants={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="background-image"
        />,
        <div
          className="app-frame"
        />,
        <div
          className="App-content"
        >
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Form)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            exact={true}
            path="/"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(CardContainer)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            exact={true}
            path="/locations"
          />
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "background-image",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "app-frame",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Form)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              exact={true}
              path="/"
            />,
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(CardContainer)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              exact={true}
              path="/locations"
            />,
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
          ],
          "className": "App-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Form)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(CardContainer)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "exact": true,
              "path": "/locations",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="background-image"
          />,
          <div
            className="app-frame"
          />,
          <div
            className="App-content"
          >
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Form)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              exact={true}
              path="/"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(CardContainer)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              exact={true}
              path="/locations"
            />
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "background-image",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "app-frame",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(Form)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                exact={true}
                path="/"
              />,
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(CardContainer)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                exact={true}
                path="/locations"
              />,
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
            ],
            "className": "App-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Form)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(CardContainer)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "exact": true,
                "path": "/locations",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
